### PR TITLE
Significantly speed up casecheck.py

### PR DESCRIPTION
**Pull request by Bobrosoft98**
Make time complexity of casecheck.py linear (using set)

### CODE REVIEW

**Comment by cr** [comment]
Looks like I was wrong to assume that I/O would be the bottleneck. Getting rid of the quadratic complexity by using a set yields a 200% speed-up on a random test tree from my SSD:

```
In [2]: %timeit without_set()
1 loop, best of 3: 45.1 s per loop

In [3]: %timeit with_set()
1 loop, best of 3: 13.6 s per loop
```

File: casecheck.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
...
 	content = dirs + files
-	for one in content:
-		for other in content:
-			if one != other:
-				if one.lower() == other.lower():
-					print(root + "/" + one)
+	all_filenames = set()
+	for filename in content:
+		if filename.lower() in all_filenames:
+			print(root + "/" + one)
```
Comment (cr) [inline casecheck.py]: This fails when a collision is found. Replace ```one``` with ```filename```.

**Review by cr — CHANGES_REQUESTED** [review]
Thanks for the PR! Please fix the bug and I'll happily land it.

**Review by Bobrosoft98 — COMMENTED** [review]
Sorry for that. :) Tested it locally, but decided to change "one" to "filename" at the last moment. I'm a bit confused with the GitHub interface for pull requests, but looks like I've fixed that bug.

**Review by cr — APPROVED** [review]

**Comment by cr** [comment]
Right, just keep pushing onto your branch until everyone's merry.

Squashed and merged. Thanks for bringing it up!

**Pull request merged**